Updated the security card notice design.

zinigor opened the pull request:
Fixes #6405.

#### Changes proposed in this Pull Request:
* Adds a property to the SettingsCard element to allow passing a notice in.
* Adds a notice when the user has no plan, leading them to ugrade:
![upgrade](https://cloud.githubusercontent.com/assets/374293/23082873/afaecbc4-f574-11e6-80fb-6d7fb626379e.png)
* Adds a notice when the user has a plan but the scanning is not active:
![activate](https://cloud.githubusercontent.com/assets/374293/23082895/ce097c40-f574-11e6-9f40-13b8f5c09d4a.png)
* Adds a notice when the user has a plan, has scan active and has threats:
![threats](https://cloud.githubusercontent.com/assets/374293/23082929/e7479142-f574-11e6-988a-5710ad291660.png)
* Adds a notice when the user has a plan, has scan active and no threats:
![ok](https://cloud.githubusercontent.com/assets/374293/23082966/0da0cf02-f575-11e6-9d49-45b1c03f1c06.png)

This PR copies a lot of logic that is already used by the At a Glance widget. Once this is merged I will create an issue about refactoring both of them and using a memoized selector to get rid of code duplication and simplify data retrieval.

dereksmart commented:
This is failing both lint and GUI tests.  

Linting errors: 
```
/_inc/client/security/backups-scan.jsx
error    'hasSitePlan' is assigned a value but never used            no-unused-vars
error    'inactiveOrUninstalled' is assigned a value but never used  no-unused-vars
```

GUI failing test 
```
 1) SettingsCard the heading has the right text:
      AssertionError: expected '' to equal 'Comments'
      + expected - actual
      +Comments
      
      at Context.<anonymous> (_inc/client/components/settings-card/test/component.js:38:65)
```

zinigor commented:
@dereksmart thanks for testing, I have pushed two new commits that fix the build problems.

eliorivero commented on _inc/client/components/settings-card/index.jsx:
```diff
@@ -10,42 +10,58 @@ import { translate as __ } from 'i18n-calypso';
 import SectionHeader from 'components/section-header';
 import Button from 'components/button';
 
-const SettingsCard = props => {
```
As discussed in Slack, this should be restored as a stateless component since no component state methods provided by React.createClass are used.

eliorivero commented on _inc/client/components/settings-card/index.jsx:
```diff
@@ -10,42 +10,58 @@ import { translate as __ } from 'i18n-calypso';
-			: '',
-		isSaving = props.isSavingAnyOption();
+export default React.createClass( {
+	propTypes: {
```
Props for adding types and defaults 🎉 👍

eliorivero commented:
Still getting the same warnings

zinigor commented:
@eliorivero I have cleaned up the props, you should no longer see the warnings. @dereksmart the problems should go away after I have rebased the PR. Thanks for reviewing both of you!

eliorivero commented:
Warnings are gone now! 👍 

One issue: the notice is showing in dev mode and it shouldn't.

dereksmart commented:
> This PR copies a lot of logic that is already used by the At a Glance widget.

Were you not able to re-use/refactor the `<ProStatus` component to do this instead?  It's already used in the AAG widget and was re-used by the old settings cards.

dereksmart commented on _inc/client/security/backups-scan.jsx:
```diff
@@ -12,37 +14,127 @@ import ExternalLink from 'components/external-link';
+			link = false;
+		} else {
+			status = 'is-warning';
+			text = __( 'Upgrade', { context: 'Caption for a button to purchase a paid feature.' } );
```
Do we even want to show this?  We're going to have the banner below the card as well.

zinigor replied:
That's a good point, @MichaelArestad you mentioned that we need to hide the card when it's not functioning, do we hide the notice when there's an upgrade banner as well?

zinigor replied:
So what I meant to say is that you're right @dereksmart . This condition branch is not going to be needed because we're going to be hiding the card in this case.

dereksmart replied:
@zinigor so will it be removed in this branch?

zinigor replied:
Yes, I have refactored the whole thing in a newer commit, but it looks like I didn't push it yet. I was trying to give more attention to PRs that may be going into master last week and today. I'm going to push these changes as soon as they are ready.

MichaelArestad commented:
@zinigor is this ready for review?

MichaelArestad commented:
Let's change the "All clean" label to "Secure"

MichaelArestad commented:
It's also not vertically centered. Is that a problem with the other ones we use in DashItems?

zinigor commented:
@eliorivero thanks, I have added a condition that stops the notice from being displayed in dev mode.
@MichaelArestad yes, it's ready for review, I have pushed several new commits.
@dereksmart thanks for the review, I have repurposed the `ProStatus` component to work both here and in the dashboard.

eliorivero commented:
Found a couple of issues:

1. when VP plugin is deactivated, there's a blue SET UP button. I found no way to display the yellow "Inactive FIX IT" notice.

<img width="729" alt="captura de pantalla 2017-03-07 a las 15 26 22" src="https://cloud.githubusercontent.com/assets/1041600/23671607/a280731a-034a-11e7-8205-852cd1dc75fa.png">

2. the "Upgrade FIX IT" notice feels very awkward. Upgrade what? Fix what? Maybe it should be some word reflecting the limitations of the feature in the current plan. Maybe "Basic Protection UPGRADE"? What do you think @richardmuscat?

3. when the app is loading, the card looks like this. I understand that it needs a bit of time to fetch the data. Maybe we could not display the banner until data has been received and we know for sure if we're displaying it or not?

<img width="764" alt="captura de pantalla 2017-03-07 a las 15 41 59" src="https://cloud.githubusercontent.com/assets/1041600/23672220/b7f2fd24-034c-11e7-8d15-b37fc5207abb.png">

zinigor commented:
I have rebased the PR, but there is a problem. This PR changes `SettingsCard` usage so that:
* we no longer pass `{ ...this.props}` to it, but instead pass only needed props
* no longer retrieve the module from the passed `getModule` prop, only use whatever is passed in the `module` prop;
* hence the `module` is always expected to be an object, not a slug.

This creates problems with some recent changes made in `feature/settings-overhaul` where sometimes we still pass slugs in as `module` prop, they all need to be changed from `module="something"` to `module={ { module: "something" } }`.

eliorivero commented:
Closed in favor of #6687
